src/core/ioc.py

```diff
@@ -0,0 +1,77 @@
+# structure for the ioc to have just one instance
+
+from src.core.singleton_decorator import singleton
+from src.lib.repositories.impl import *
+from src.api.controllers import *
+
+
+@singleton
+class Ioc:
+
+    _instance_ioc = {}
```

Comment:
Could initialize this within `__init__`?

Reply:
No, if i do that the variable will belong to the object instance not to the class and to be singleton i want the varible to belong the class thus the variable will be share for all instances.